### PR TITLE
Add Docker Hub Webhook Plugin

### PR DESCRIPTION
**Pull request by DazWilkin**
### Required for all PRs:
- [ ] CHANGELOG.md updated
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [X] README.md updated (if adding a new plugin)

### CODE REVIEW

**Comment by danielnelson** [comment]
@DazWilkin, sorry about the delay on this pull request.  Would you be able to rebase your pull request to resolve the conflicts?  If so, I can review and we should be able to add this for [1.6](https://github.com/influxdata/telegraf/milestone/22)

File: plugins/inputs/webhooks/dockerhub/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+An event is generated by Docker Hub as the result of a docker push to the repository.  
+
+#### [`dockerhub_event`](https://docs.docker.com/docker-hub/webhooks/)
+
+The measurement is called "dockerhub"
+
+**Tags:**
+
+* 'description' = `repository.description` string
```
Comment (glinton) [inline plugins/inputs/webhooks/dockerhub/README.md]: Probably don't want to index "description" as a tag, make it a field.

File: plugins/inputs/webhooks/dockerhub/dockerhub_webhooks.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+type DockerhubWebhook struct {
+	Path string
+	acc  telegraf.Accumulator
+}
+
+func (dhwh *DockerhubWebhook) Register(router *mux.Router, acc telegraf.Accumulator) {
+	router.HandleFunc(dhwh.Path, dhwh.eventHandler).Methods("POST")
+	log.Printf("Started '%s' on %s\n", meas, dhwh.Path)
```
Comment (glinton) [inline plugins/inputs/webhooks/dockerhub/dockerhub_webhooks.go]: Make this a debug log or remove.

File: plugins/inputs/webhooks/dockerhub/dockerhub_webhooks.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+func (dhwh *DockerhubWebhook) eventHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	e, err := NewEvent(data, &DockerhubEvent{})
```
Comment (glinton) [inline plugins/inputs/webhooks/dockerhub/dockerhub_webhooks.go]: Why not just do the `json.Unmarshal` here instead.

File: plugins/inputs/webhooks/dockerhub/dockerhub_webhooks_mocks.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+func RandStringBytes(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = hexBytes[rand.Intn(len(hexBytes))]
+	}
+	return string(b)
+}
+
+func NewEventJSONEncoded() string {
```
Comment (glinton) [inline plugins/inputs/webhooks/dockerhub/dockerhub_webhooks_mocks.go]: Put this all in the `_test.go` file

**Comment by russorat** [comment]
Closing this due to inactivity. If you are interested in this, please comment.

**Pull request closed without merging**